### ISSUE TITLE
Iris handles datapack dimensions in a way that is inconsistent with both Vanilla and OptiFine

### ISSUE THREAD

**Issue by TBlazeWarriorT** [IrisShaders/Iris#1227]
### What happened?

- Iris (and OptiFine, for reference) fail to render datapack dimensions even in "vanilla MC" in a way that minimally looks how it does and should look like in vanilla, including but not limited to dimension fixed time, sky effects, sky light and ambient light
- Iris and OptiFine dimension detection is inconsistent with each other, with (I think) OptiFine handling it better - (dimension type was set to "the_overworld" but it only had a curvature on optifine, something else was making Iris think it was The End. The last screenshot are a custom end dimension, but it is based on the overworld so that it looks less broken with shaders, so technically Iris should have probably detected it as such)
- Iris and OptiFine have drastically different visuals on the same place and settings. I guess one of the two has to be wrong?

### Screenshots

Dimension Settings for the first 3 screenshots:
daytime: 6000
fixed dimension time: none
sky effects: the_end
skylight: true
ambient light: 0
overworld curvature: 1024
end curvature: 0 

OptiFine with Shaders Off (reference image)
![image](https://user-images.githubusercontent.com/25572454/150361321-eba0149e-befb-4ea6-8e68-faa44cb837ac.png)
OptiFine with Complementary (reference image)
![image](https://user-images.githubusercontent.com/25572454/150361546-7f8d02a3-c592-4165-bccd-61a0e5acf811.png)
Iris with Complementary
![image](https://user-images.githubusercontent.com/25572454/150361598-1e42221d-b402-451a-8516-47d1fb8d4440.png)
Iris 1.18 (the datapack does not work in 1.18 so the biome is different, I just wanted to prove that it would look more or less the the same in 1.18 and wasn't fixed)
![image](https://user-images.githubusercontent.com/25572454/150364122-d2458bb2-0706-43aa-8c36-ca9a47b7ebc9.png)

Different-ish setup with Dimension Type set to Overworld and Daytime set to Noon (lots of curvature and lighting/color differences - exact same place and settings on both, only left has curvature)
![image](https://user-images.githubusercontent.com/25572454/150361756-b12c22ec-321a-4df6-87fd-7d4e09f667ca.png)

### Minecraft Version

1.16.5, 1.18.1

### Iris Version

iris-mc1.16.5-1.2.0-pre.jar

### Sodium Version

sodium-fabric-mc1.16.5-0.2.0+build.4.jar

### Operating System

Windows 10

### What is your GPU?

RTX 2060

### Java Version

Bundled

### Additional context

Not exactly a bug but almost definitely not working how it should or was intended to look like in a perfect scenario. Can cause quite a bit of annoyance and might be hiding one or more bugs behind it.

I tried to include a few decent screenshots but there are just too many variables and having to switch between Iris and OptiFine to compare made it too confusing and hard to track. Hopefully the 3 "issues" above (which kinda are the same) are explained in a decent and helpful way still.

**Comment by coderbot16** [comment]
https://github.com/IrisShaders/Iris/commit/7529fea14e261b0f896b6a4808814c5f90882ef1 might fix some of these issues.

**Comment by coderbot16** [comment]
@TBlazeWarriorT Can you test again on 1.2.1 / 1.2.2?

**Comment by TBlazeWarriorT** [comment]
@coderbot16 do I need to fully redo the testing?
I just went to the same place in 1.16.5 1.2.2 and after a few seconds I realized it looks great
Overall look: good
Dimension Detection for World Curvature: End (good)
Ambient Light display: broken (0 and 15 looks identical when shaders are enabled)
Fixed Time: does nothing (probably good, end doesn't have daytime?)
Normal Maps: working, I think it looks normal to me?

So TL;DR only ambient light seems to be still broken but almost nobody uses it
Thx a lot for making my adventure map's end dimension look gorgeous
![image](https://user-images.githubusercontent.com/25572454/158303235-f25d261e-504b-4110-91a2-a2989975d98e.png)

**Comment by coderbot16** [comment]
Great, thanks for testing, I'll close this as it seems like 1.2.1 resolved it then. Seems like @IMS212's fixes for Terralith's nullscape stuff fixed this too.

**Comment by TBlazeWarriorT** [comment]
![image](https://user-images.githubusercontent.com/25572454/158305652-89342f97-31bd-4fc1-bfcc-20ad529306f5.png)
![image](https://user-images.githubusercontent.com/25572454/158305673-041329f8-03e2-4aa3-92f0-cba6f58973c8.png)
I checked another custom dimension from the same map and realized this at the last second, is this enough to get you worried again or worth including on this issue?
Even at the default Complementary settings the sky and fog color looks a bit too different from what it should. Other shaders seem to just ignore vanilla sky/fog color so it's a bit harder to test but it only looks farther than vanilla on all other shaders I have

**Comment by coderbot16** [comment]
Ultimately that's likely just a stylistic decision by the shader - it can decide to ignore fog colors and sky colors or modify them however it wants. Complementary and BSL take the sky color into account, but they modify it to suit their style.
